src/app/components/registration-page/registration-page.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { User } from 'src/app/models/User';
+import { RegisterApiService } from 'src/app/services/restapiservice/register-api-service';
+import { RestApiServiceService } from 'src/app/services/restapiservice/rest-api-service.service';
+
+@Component({
+  selector: 'app-registration-page',
+  templateUrl: './registration-page.component.html',
+  styleUrls: ['./registration-page.component.css']
+})
+export class RegistrationPageComponent implements OnInit {
+  loginForm: FormGroup;
+  isPasswordValid:boolean = true;
+  constructor(private service :RegisterApiService) { }
+
+  ngOnInit(): void {
+    this.loginForm = new FormGroup({
+      Username: new FormControl('', [ Validators.required]),
+      Password: new FormControl('', [ Validators.required]),
+      ConfirmPassword: new FormControl('', [ Validators.required]),
+      Email: new FormControl('', [Validators.required, Validators.email]),
+    });
+  }
+
+  register(){
+    var user = new User();
+    if(this.loginForm.get('Password').value == this.loginForm.get('ConfirmPassword').value){
+      this.isPasswordValid =  true;
+      var user = new User(this.loginForm.get("Username").value,this.loginForm.get("Password").value,this.loginForm.get("Email").value)
```

Comment:
In loc de new User(parameters)
se poate face direct let user: User = this.loginForm.value sau ceva de genul

Reply:
Sau creezi un obiect gol

const registerRequest = {
username: username,
password: password
}